policy/modules/system/systemd.if

```diff
@@ -58,28 +69,603 @@ template(`systemd_role_template',`
 	# This domain is per-role because of the below transitions.
 	# See the systemd --user section of systemd.te for the
 	# remainder of the rules.
-	allow $1_systemd_t $3:process { setsched rlimitinh };
+	allow $1_systemd_t $3:process { setsched rlimitinh signal_perms };
+	allow $1_systemd_t self:netlink_kobject_uevent_socket create_socket_perms;
 	corecmd_shell_domtrans($1_systemd_t, $3)
 	corecmd_bin_domtrans($1_systemd_t, $3)
-	allow $1_systemd_t self:process signal;
+	allow $1_systemd_t self:process { getsched signal };
 
 	files_search_home($1_systemd_t)
 
+	init_link_key_template($1_systemd_t)
+
+	miscfiles_watch_localization($1_systemd_t)
+
+	mount_read_runtime_files($1_systemd_t)
+	mount_watch_runtime_files($1_systemd_t)
+	mount_watch_reads_runtime_files($1_systemd_t)
+
+	allow $3 $1_systemd_t:process signal;
 	# Allow using file descriptors for user environment generators
 	allow $3 $1_systemd_t:fd use;
 	allow $3 $1_systemd_t:fifo_file rw_inherited_fifo_file_perms;
+	allow $3 $1_systemd_t:unix_stream_socket { connectto rw_socket_perms };
+
+	allow $1_systemd_t self:unix_stream_socket { connectto rw_socket_perms };
+
+	allow $1_systemd_t self:dbus { acquire_svc send_msg };
+	allow $3 $1_systemd_t:dbus send_msg;
+	allow $1_systemd_t $3:dbus send_msg;
+
+	allow $3 $1_systemd_t:system { disable enable reload start stop status };
 
 	# systemctl --user
 	stream_connect_pattern($3, systemd_user_runtime_t, systemd_user_runtime_t, $1_systemd_t)
 
+	search_dirs_pattern($1_systemd_t, $3, $3)
+	read_files_pattern($1_systemd_t, $3, $3)
+
```

Comment:
Thats ugly but the search_dirs_pattern is redundant as well. if i would do this then i would do it like this:

allow $1_systemd_t $3:dir search_dir_perms;
allow $1_systemd_t $3:file read_file_perms;
allow $1_systemd_t $3:lnk_file read_lnk_file_perms;

Reply:
or `ps_process_pattern`

Reply:
OK. I went with the former.